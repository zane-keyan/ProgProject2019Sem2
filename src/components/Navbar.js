import React, { Component } from "react";
import { Link } from "react-router-dom";
import icon from "../images/icon-black.png";
import { connect } from "react-redux";
import { logout } from "../store/actions/authActions";
import PropTypes from "prop-types";

class NavBar extends Component {
  static propTypes = {
    logout: PropTypes.func.isRequired
  };

  render() {
    return (
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <div className="container">
          <Link className="navbar-brand" to="/">
            <img
              alt="icon"
              src={icon}
              className="d-inline-block align-top"
              width="30"
              height="30"
            />
            {"  "}
            Homy's car
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav ml-auto">
              <Link className="nav-item nav-link active" to="/">
                HOME <span className="sr-only">(current)</span>
              </Link>
              <h2 className="d-none d-lg-block my-h2">
                &nbsp;&nbsp;/&nbsp;&nbsp;
              </h2>
              <Link className="nav-item nav-link" to="/about">
                ABOUT
              </Link>

              {this.displayAuthLink(this.props.isAuthenticated)}
            </div>
          </div>
        </div>
      </nav>
    );
  }
  displayAuthLink = isAuthenticated => {
    if (isAuthenticated) {
      return (
        <React.Fragment>
          <h2 className="d-none d-lg-block my-h2">&nbsp;&nbsp;/&nbsp;&nbsp;</h2>
          <Link
            className="nav-item nav-link"
            onClick={this.props.logout}
            to="/"
          >
            {" "}
            LOGOUT
          </Link>
        </React.Fragment>
      );
    } else {
      return (
        <React.Fragment>
          {" "}
          <h2 className="d-none d-lg-block my-h2">&nbsp;&nbsp;/&nbsp;&nbsp;</h2>
          <Link className="nav-item nav-link" to="/signin">
            SIGN IN
          </Link>
          <h2 className="d-none d-lg-block my-h2">&nbsp;&nbsp;/&nbsp;&nbsp;</h2>
          <Link className="nav-item nav-link" to="/signup">
            SIGN UP
          </Link>
        </React.Fragment>
      );
    }
  };
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});
export default connect(
  mapStateToProps,
  { logout }
)(NavBar);
