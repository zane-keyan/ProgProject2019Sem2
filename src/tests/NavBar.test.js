import React from "react";
import renderer from "react-test-renderer";
import Navbar from "../components/Navbar";
import { BrowserRouter as Router } from "react-router-dom";

it("render correctly Navbar Component", () => {
  const NavbarComponent = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();
  expect(NavbarComponent).toMatchSnapshot();
});
