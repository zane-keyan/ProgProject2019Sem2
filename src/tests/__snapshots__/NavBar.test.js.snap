// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Navbar Component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="icon"
        className="d-inline-block align-top"
        height="30"
        src="icon-black.png"
        width="30"
      />
        
      Homy's car
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav ml-auto"
      >
        <a
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
        >
          HOME 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
        <h2
          className="d-none d-lg-block my-h2"
        >
            /  
        </h2>
        <a
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          ABOUT
        </a>
        <h2
          className="d-none d-lg-block my-h2"
        >
            /  
        </h2>
        <a
          className="nav-item nav-link"
          href="/signin"
          onClick={[Function]}
        >
          SIGN IN
        </a>
        <h2
          className="d-none d-lg-block my-h2"
        >
            /  
        </h2>
        <a
          className="nav-item nav-link"
          href="/signup"
          onClick={[Function]}
        >
          SIGN UP
        </a>
      </div>
    </div>
  </div>
</nav>
`;
